### ISSUE TITLE
JsonObject Boolean issue

### ISSUE THREAD

**Issue by jpaxy** [awslabs/aws-crt-cpp#200]
Hello, 

I noticed today that set/get functionality of Aws::Crt::JsonObject does not work as expected.

This happens e.g. with following example when I try to replace existing value with new. Value does not change.
```
bool Configuration::setBoolValue(const Aws::Crt::String key, bool value)
{
    *m_configJson = m_configJson->WithBool(key, value);
    return m_configJson->View().GetBool(key));
}
```
With Intergers, Strings etc this works fine.

**Comment by stwirth** [comment]
I encountered the same issue today:

```C++
TEST(JsonObject, booltest)
{
  Aws::Crt::JsonObject object;
  object.WithBool("my_true_bool", true).WithBool("my_false_bool", false);
  std::cout << "Serialized: " << std::endl;
  std::cout << object.View().WriteReadable().c_str() << std::endl;

  /** prints
   {
   	"my_true_bool":	true,
   	"my_false_bool":	false
   }
  */

  EXPECT_TRUE(object.View().GetJsonObject("my_false_bool").IsBool());  // pass
  EXPECT_FALSE(object.View().GetJsonObject("my_false_bool").AsBool()); // pass
  EXPECT_FALSE(object.View().GetBool("my_false_bool")); // pass

  EXPECT_TRUE(object.View().GetJsonObject("my_true_bool").IsBool()); // pass
  EXPECT_TRUE(object.View().GetJsonObject("my_true_bool").AsBool()); // pass
  EXPECT_TRUE(object.View().GetBool("my_true_bool"));  // fail ?!?!
}
```
`GetBool(key)` gives a different result than `GetJsonObject(key).AsBool()`.

**Comment by bretambrose** [comment]
Thanks for finding this; it should be fixed in v0.13.7
